validate_test.go

```diff
@@ -0,0 +1,97 @@
+package gitlab
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"reflect"
+	"testing"
+)
+
+// XXX: move this func out if necessary
+// content read a file, return its content as string
+func content(f string) string {
+	b, err := ioutil.ReadFile(f)
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+	c := string(b)
+	return c
+}
+
+// TODO: better test posting empty string
+func TestValidate(t *testing.T) {
+	mux, server, client := setup()
+	defer teardown(server)
+
+	var res string
+
+	valid := `{
+			"status": "valid",
+			"errors": []
+		}`
+
+	invalid := `{
+			"status": "invalid",
+			"errors": [
+				"variables config should be a hash of key value pairs"
+			]
+		}`
+
+	mux.HandleFunc("/ci/lint", func(w http.ResponseWriter, r *http.Request) {
+		testMethod(t, r, "POST")
+		fmt.Fprintf(w, res)
+	})
+
+	testFunc := func(t *testing.T, got, want *LintResult) {
```

Comment:
@svanharmelen Not sure how you feel about bringing in a test dependency. Would suggest replacing `testFunc` with a call to testify/assert
```
assert.Equal(t, expected, actual)
```
Ref: https://github.com/stretchr/testify

Reply:
Not sure. Not really a fan of using external packages for writing tests (especially for client packages like this).

About the `testFunc` specifically, I think it should be factored out once using table driven tests.